### DISCUSSION TITLE
On the algorithm problem of post-inference merging

### DISCUSSION THREAD

**Discussion by gucunlin** [Project-MONAI/MONAI#7848] [Q&A]
"I am porting a model used in Monai inference to C++ using ITK and libtorch. However, I encounter a problem when merging the blocks after inference; the seams are too obvious, and the data differs significantly from that merged in Monai. For example, when merging 100 blocks of size 96, 96, 96 with an overlap rate of 62.5%, the seams are visibly prominent, and the final results after softmax and argmax do not match those in Python. How can I solve this issue with the overly obvious seams?"
![微信截图_20240614171535](https://github.com/Project-MONAI/MONAI/assets/37055528/154b5b62-2999-4707-9a74-f105a685cdf7)

**Comment by KumoLiu** [comment]
Hi @gucunlin, thanks for your interest here.

The issue you're encountering with merging the blocks post-inference typically arises due to the way the overlapping predictions are combined. In MONAI, seamless tiling is often achieved by using methods that blend the overlapping regions to ensure continuity between blocks. Here are a few strategies you can try to solve the issue of obvious seams:
- **Overlap-tile Strategy Adjustment**: First, ensure your overlap rate is adequate. Your current rate of 62.5% might be causing the blocks to blend incorrectly. Experiment with different overlap rates to see if it benefits the merging.
- **Weighted Blending**: When combining the blocks, you can use a weighted blending strategy, where the contributions of the overlapping predictions are weighted according to their distance from the edge of the block. Blocks closer to the center of an overlap can be given more weight than those closer to the edges.

Hope it helps, thanks.

**Reply by gucunlin** [reply]
think you, I'm finish it. Problems with merging functions
